google/cloud/aiplatform/models.py

```diff
@@ -15,9 +15,213 @@
 # limitations under the License.
 #
 
+from google.cloud.aiplatform_v1beta1.services.model_service import (
+    client as model_client,
+)
 
-class Model:
-    pass
+from typing import Dict, Optional, Sequence
+
+from google.auth import credentials as auth_credentials
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform import initializer
+from google.cloud.aiplatform import utils
+from google.cloud.aiplatform_v1beta1.services.model_service.client import (
+    ModelServiceClient,
+)
+from google.cloud.aiplatform_v1beta1.types import model as gca_model
+from google.cloud.aiplatform_v1beta1.types import env_var
+
+
+class Model(base.AiPlatformResourceNoun):
+
+    client_class = ModelServiceClient
+    is_prediction_client = False
```

Comment:
Semantically this can be a little confusing, as if we are saying the Model class is prediction client: `Model.is_prediction_client`, while we mean `Model.client.is_prediction_client`.

Reply:
Renaming to _is_client_prediction_client

Reply:
Done.